src/main/dynamo/types.ts

```diff
@@ -11,3 +11,5 @@ export type ExtractKeyType<T> = T extends PartitionAndSortKey<infer U>
 export type GroupedModels<T extends TaggedModel> = {
   [K in T["model"]]: T extends { model: K } ? T[] : never
 }
+
+export type AtLeastOne<T, U = { [K in keyof T]: Pick<T, K> }> = Partial<T> & U[keyof U]
```

Comment:
Hmm... This ends up generating a pretty crazy type that's hard to read. And it appears to be basically equivalent to `Partial<{ [X in U]: string }>` -- i.e. you can specify one or more of the key parts.